### PR TITLE
[CDAP-16690] Add record view when number of fields is large

### PR DESCRIPTION
**Pull request by yukiej**
JIRA: https://issues.cask.co/browse/CDAP-16690
Build: https://builds.cask.co/browse/CDAP-UDUT679

- Added row numbers for data table view
- Added record view for when number of fields in schema is large (currently 100 or greater, but this can be changed easily). 
- Can navigate records using record dropdown or by clicking forward/backward. 
- If there are multiple inputs or outputs, will show each stage's records in its own tab

[An example pipeline with 100+ fields in the sources and joiner (but not the sink, so you should see the table view for the sink)](https://drive.google.com/open?id=1kAJjO_Eg7L62-MuzLngA_4zxpy0580g2) -- need to update the service file path for the cdap-gcp-project!

Screen Shot 2020-06-17 at 7 29 55 PM

record_view

### CODE REVIEW

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx [hunk 1]
```diff
@@ -64,6 +70,7 @@ const styles = (theme) => ({
     },
   },
   messageText: messageTextStyle,
+  rowNumber: {},
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: is this empty styles needed?

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx [hunk 2]
```diff
@@ -110,6 +117,7 @@ const DataTableView: React.FC<IDataTableProps> = ({
       <Table>
         <TableHead>
           <TableRow className={classes.row}>
+            <CustomTableCell></CustomTableCell>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: nit: `<CustomTableCell />`

File: cdap-ui/app/cdap/components/PreviewData/DataView/TableContainer.tsx [hunk 3]
```diff
@@ -43,6 +44,7 @@ const styles = (theme): StyleRules => ({
     '& :last-of-type': {
       borderRight: 0,
     },
+    '& .record-pane': { width: '100%' },
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/DataView/TableContainer.tsx:47]: can this width `100%` be on the actual component itself (preventing the use of named class)?

File: cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  nextOperation,
+}) => {
+  const prevDisabled = selectedRecord - 1 < 1;
+  const nextDisabled = selectedRecord + 1 > numRecords;
+
+  const selectOptions = [];
+  for (let i = 1; i <= numRecords; i++) {
+    const value = `Record ${i}`;
+    selectOptions.push({ value, label: value });
+  }
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx]: for initialization type code, especially if it contains loops is better to be contained in a `useEffect` so it only executes when it needs to

File: cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          widgetProps={{ options: selectOptions }}
+          onChange={(e) => updateRecord(e)}
+        />
+      </span>
+      <span
+        className={classnames('nav-icon', { disabled: nextDisabled })}
+        onClick={!nextDisabled ? nextOperation : undefined}
+      >
+        <IconSVG name="icon-caret-right" />
+      </span>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx]: can you investigate to see if perhaps using material-ui button icons would be easy to style? this would prevent additional disabled styling and onClick handling

File: cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+      <div className={classes.outerContainer}>
+        <If condition={!selectedNode.isSource && !selectedNode.isCondition}>
+          <div
+            className={classnames(classes.innerContainer, {
+              [classes.split]: !selectedNode.isSource && !selectedNode.isSink,
+            })}
+          >
+            <h2 className={classes.h2Title}>{T.translate(`${I18N_PREFIX}.inputHeader`)}</h2>
+            {showInputTabs
+              ? getTabs(getTabConfig(inputs, selectedRecord))
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx:107]: edge case to test would be how overflow will be handled for the tabs (if there are a lot of inputs our outputs)

File: cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                    <RecordTable
+                      headers={stageInfo.schemaFields}
+                      record={stageInfo.records[selectedRecord - 1]}
+                    />
+                  );
+                })}
+          </div>
+        </If>
+        <If condition={!selectedNode.isSink && !selectedNode.isCondition && showOutputTabs}>
+          <div></div>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx]: is this empty intentional or should there be some message?

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx (hunk 1 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: Good catch! Removed

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx (hunk 2 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: Updated

File: cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx (hunk 7 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx]: This shouldn't be there at all - I've removed this If component.

File: cdap-ui/app/cdap/components/PreviewData/DataView/TableContainer.tsx (hunk 3 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/DataView/TableContainer.tsx:47]: I don't think so - this is here because the 100% width needs to be applied to the tab-pane, and I couldn't figure out another way to style the element nested inside the tab.

File: cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx (hunk 6 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx:107]: Good call out! I've added styling to make sure the tabs scroll if there is overflow.

File: cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx (hunk 4 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx]: That is a good call. Moved to a useEffect!

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx (hunk 1 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: Nope! Removed

File: cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx (hunk 2 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx]: Fixed!

File: cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx (hunk 5 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx]: Used material-ui IconButton component instead.

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 8]
```diff
@@ -2413,14 +2413,19 @@ features:
...
     runPreview: Run preview to generate preview data.
-    Table: 
-      noPreviewRunning: "{recordType} records have not been generated. Please verify your logic or try sending more data."
-      previewNotSupported: Preview data is not supported for condition stages.
-      previewRunning: "{recordType} records have not been generated yet. Please check again in a few minutes."
-    TableContainer:
-      conditionHeader: Input Records and Output Records
-      inputHeader: Input Records
-      outputHeader: Output Records
+    DataView:
```
Comment (elfenheart) [inline cdap-ui/app/cdap/text/text-en.yaml:2413]: nit: sort alphabetically

**Review by elfenheart — APPROVED** [review]
One nit, otherwise 👍  LGTM
